specs/ExtendedNewWindowRequested.md

```diff
@@ -0,0 +1,78 @@
+# Background
+WebView2 raises the `NewWindowRequested` event when a new window is opened. The WebView2 team has been asked to include the name of the new window 
+being created as a parameter of `NewWindowRequestedEventArgs`. This window name is the name given to the window when it is opened 
+using `window.open(url, windowName)`.
+
+In this document we describe the updated API. We'd appreciate your feedback.
+
+# Description
+We propose extending the `NewWindowRequestedEventArgs` to provide access to the `Name` property. 
+The `Name` property will return the name of the new window being created. 
+
+# Examples
+## C++: Get name of window
+
+``` cpp
+wil::com_ptr<ICoreWebView2> m_webviewEventSource;
+EventRegistrationToken m_newWindowRequestedToken = {};
+
+m_webviewEventSource->add_NewWindowRequested(
+    Callback<ICoreWebView2NewWindowRequestedEventHandler>(
+        [this](ICoreWebView2* sender, ICoreWebView2NewWindowRequestedEventArgs* args)
+          -> HRESULT {
+              wil::com_ptr<ICoreWebView2NewWindowRequestedEventArgs2> args2;
+              
+              if(SUCCEEDED(args->QueryInterface(IID_PPV_ARGS(&args2)))) {
+                  wil::unique_cotaskmem_string name;
+                  CHECK_FAILURE(args2->get_Name(&name));
+              }
+
+              return S_OK;
+        })
+        .Get(),
+  &m_newWindowRequestedToken);
+```
+
+## C#: Get name of window
+```c#
+webView.CoreWebView2.NewWindowRequested += WebView_NewWindowRequested;
+
+void WebView_NewWindowRequested(object sender, CoreWebView2NewWindowRequestedEventArgs e)
+{
+    string newWindowName = e.Name;
+    MessageBox.Show(this, newWindowName);
```

Comment:
Similarly here it would be good to show how to use the name in the way the customer is asking about it.

```suggestion
    if (newWindowName == "openInNewWindow")
    {
        Process.Start(e.Url);
        e.Handled = true;
    }
    else
    {
        // normal new window handling behavior
    }
```